aspen/request_processor/algorithm.py

```diff
@@ -70,6 +70,11 @@ def load_resource_from_filesystem(request_processor, dispatch_result):
     return {'resource': resources.get(request_processor, dispatch_result.match)}
 
 
-def render_resource(state, request_processor, resource):
-    state.setdefault('output', output.Output(charset=request_processor.charset_dynamic))
+def render_resource(state, resource):
     return {'output': resource.render(state)}
+
+
+def encode_output(output, request_processor):
+    if isinstance(output.body, unicode):
+        output.charset = request_processor.charset_dynamic
```

Comment:
> Maybe something like `encode_dynamic_resources_as` and `assume_static_resources_are_encoded_as`?

A bit lengthy, isn't it? How about leaving `charset_static` as is and renaming `charset_dynamic` to `encode_output_as`?